onnx/funconnx/proto_run.py

```diff
@@ -0,0 +1,368 @@
+        y = Celu.eval(x, alpha=0.5)
+        print(y)
+
+    Method :meth:`eval <onnx.ProtoRun.op_run.eval>` creates an onnx node
+    returned by method :meth:`make_node <onnx.funconnx.op_run.make_node>`.
+
+    .. exec_code::
+
+        import numpy as np
+        from onnx.funconnx.aionnx._op_list import Celu
+
+        onnx_node = Celu.make_node(alpha=0.5)
+        print(onnx_node)
+    """
+
+    def __init__(  # type: ignore
+        self,
+        proto: Any,
+        opsets: Union[None, Dict[str, int]] = None,
+        functions=None,
+        verbose: int = 0,
+        new_ops: Optional[List[OpRun]] = None,
+    ):
+        self.output_types_ = None
+        self.input_types_ = None
+        if isinstance(proto, str):
+            with open(proto, "rb") as f:
+                proto = load(f)
+        elif isinstance(proto, bytes):
+            proto = load(BytesIO(proto))
+        self.proto_ = proto
+        self.functions_: Dict[Tuple[str, str], ProtoRun] = {}
+        self.attributes_: List[str] = []
+        if isinstance(proto, ModelProto):
+            self.onnx_graph_ = proto.graph
+            self.opsets_ = {d.domain: d.version for d in proto.opset_import}
+            if opsets is not None:
+                raise ValueError("opsets must be None if proto is ModelProto.")
+            if functions is not None:
+                raise ValueError("functions must be None if proto is ModelProto.")
+            functions = proto.functions
+        elif isinstance(proto, GraphProto):
+            self.onnx_graph_ = proto
+            if not isinstance(opsets, dict):
+                raise ValueError("opsets must be a dictionary if proto is GraphProto.")
+            self.opsets_ = opsets
+        elif isinstance(proto, FunctionProto):
+            self.onnx_graph_ = None  # type: ignore
+            self.opsets_ = {d.domain: d.version for d in proto.opset_import}
+            if opsets is not None:
+                raise ValueError("opsets must be None if proto is FunctionProto.")
+            self.attributes_ = list(proto.attribute)
+        else:
+            raise TypeError(f"Unexpected type {type(proto)} for proto.")
+        if self.onnx_graph_:
+            self.input_names_ = [i.name for i in self.onnx_graph_.input]
+            self.input_types_ = [i.type for i in self.onnx_graph_.input]
+            self.output_names_ = [o.name for o in self.onnx_graph_.output]
+            self.output_types_ = [i.type for i in self.onnx_graph_.output]
+            self.inits_ = list(self.onnx_graph_.initializer) + list(
+                self.onnx_graph_.sparse_initializer  # type: ignore
+            )
+            self.nodes_ = self.onnx_graph_.node
+        else:
+            self.input_names_ = list(proto.input)
+            self.output_names_ = list(proto.output)
+            self.inits_ = []
+            self.nodes_ = proto.node
+        if "" not in self.opsets:
+            self.opsets[""] = onnx_opset_version()
```

Comment:
For a reference-implementation, it is better not to add "relaxations" such as the above ... it is an error if opset_imports does not include all used opsets.

Reply:
Specifically, it is important to fail with an error in any situation that the standard/spec says is an error.

Reply:
Suggest omitting lines 178-179